src/jit/MemoryInl.h

```diff
@@ -52,16 +52,32 @@ struct MemAddress {
     JITArg loadArg;
 };
 
-void MemAddress::check(sljit_compiler* compiler, Operand* offsetOperand, sljit_uw offset, sljit_u32 size)
+void MemAddress::check(sljit_compiler* compiler, Operand* offsetOperand, sljit_uw offset, sljit_u32 size, sljit_u16 memIndex)
 {
     CompileContext* context = CompileContext::get(compiler);
-    sljit_sw targetBufferOffset = context->targetBuffersStart;
 
     ASSERT(!(options & LoadInteger) || baseReg != sourceReg);
     ASSERT(!(options & LoadInteger) || offsetReg != sourceReg);
     ASSERT(!(options & CheckNaturalAlignment) || size != 1);
+    ASSERT(memIndex != (sljit_u16)-1);
+
+    MemoryType* memoryType = context->module->memoryType(memIndex);
+    uint64_t initialMemorySize = memoryType->initialSize() * Memory::s_memoryPageSize;
+    uint64_t maximumMemorySize = memoryType->maximumSize() * Memory::s_memoryPageSize;
+
+#if (defined SLJIT_32BIT_ARCHITECTURE && SLJIT_32BIT_ARCHITECTURE)
+    /* Four GB memory cannot be allocated on 32 bit systems. */
+    if (maximumMemorySize >= ((uint64_t)1 << 32)) {
+        maximumMemorySize = ((uint64_t)1 << 32) - Memory::s_memoryPageSize;
```

Comment:
I think you cannot allocate this huge memory, so this check is probably unnecessary.